src/components/MenuManager.py

```diff
@@ -283,6 +292,32 @@ def __get_page_no_to_move_to(forwards):
 
         PTLogger.debug(f"Sleep timer: {time_since_last_active}")
 
+        if self.state == MenuState.RUNNING_ACTION:
+            # Prevent dimming while running action
+            self.last_active_time = perf_counter()
+
+            time_since_action_started = perf_counter() - self.action_start_time
+            PTLogger.debug(f"Time since action started: {time_since_action_started}")
+
+            if self.current_menu.page.action_thread.is_alive():
+                PTLogger.debug("Action not yet completed")
+                return
+
+            if time_since_action_started > self.action_timeout:
+                PTLogger.info("Action timed out - setting state to WAKING")
+                self.state = MenuState.WAKING
+
+                PTLogger.info("Notifying renderer to display 'unknown' action state")
+                self.current_menu.page.hotspot.action_state = ActionState.UNKNOWN
+                return
```

Comment:
we should `join` the process before returning here, using `timeout=0` to force it